speechbrain/alignment/ctc_segmentation.py

```diff
@@ -0,0 +1,652 @@
+#!/usr/bin/env python3
+# 2021, Technische Universität München, Ludwig Kürzinger
+"""Perform CTC segmentation to align utterances within audio files."""
+
+import logging
+from pathlib import Path
+from typing import Optional
+from typing import Union
+
+import numpy as np
+import torch
+from typing import List
+
+# speechbrain interface
+from speechbrain.pretrained.interfaces import EncoderDecoderASR
+
+# imports for CTC segmentation
+try:
+    from ctc_segmentation import ctc_segmentation
+    from ctc_segmentation import CtcSegmentationParameters
+    from ctc_segmentation import determine_utterance_segments
+    from ctc_segmentation import prepare_text
+    from ctc_segmentation import prepare_token_list
+except ImportError:
+    print(
+        "ImportError: "
+        "Is the ctc_segmentation module installed "
+        "and in your PYTHONPATH?"
+    )
+
+logger = logging.getLogger(__name__)
+
+
+class CTCSegmentationTask:
+    """Task object for CTC segmentation.
+
+    When formatted with str(·), this object returns
+    results in a kaldi-style segments file formatting.
+    The human-readable output can be configured with
+    the printing options.
+
+    Properties
```

Comment:
Properties should be changed to Arguments. Please see: https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/nnet/CNN.py For the formalism. You might need to specify the type of each argument as well.

Reply:
This object is automatically generated and just a container for results, including a function for pretty printing.
There is no interface for arguments needed - I'll remove it.